crypto/cryptlib.h

```diff
@@ -106,6 +106,8 @@ extern int OPENSSL_NONPIC_relocated;
 
 char *ossl_safe_getenv(const char *);
 
+unsigned long OPENSSL_rdtsc(void);
```

Comment:
`unsigned long` is inconsistent with other declarations of this.  `uint32_t` seems like the most used variant.

Reply:
I am not sure about this: the other occurrences I see in `armcap.c` are both returning `unsigned long`.

Also changing that return type would make the signature of the function on `arm` differ across the branches.

Maybe @bernd-edlinger that contributed the original fixes can chime in about this?

Reply:
The return code of OPENSSL_rdtsc is different in 1.1.0, than what it is in 1.0.2:
```
grep -r OPENSSL_rdtsc *
crypto/modes/gcm128.c:        size_t start, stop, gcm_t, ctr_t, OPENSSL_rdtsc();
crypto/modes/gcm128.c:        start = OPENSSL_rdtsc();
crypto/modes/gcm128.c:        gcm_t = OPENSSL_rdtsc() - start;
crypto/modes/gcm128.c:        start = OPENSSL_rdtsc();
crypto/modes/gcm128.c:        ctr_t = OPENSSL_rdtsc() - start;
crypto/modes/gcm128.c:            start = OPENSSL_rdtsc();
crypto/modes/gcm128.c:            gcm_t = OPENSSL_rdtsc() - start;
crypto/pariscid.pl:	.EXPORT	OPENSSL_rdtsc,ENTRY
crypto/pariscid.pl:OPENSSL_rdtsc
crypto/x86_64cpuid.pl:.globl	OPENSSL_rdtsc
crypto/x86_64cpuid.pl:.type	OPENSSL_rdtsc,\@abi-omnipotent
crypto/x86_64cpuid.pl:OPENSSL_rdtsc:
crypto/x86_64cpuid.pl:.size	OPENSSL_rdtsc,.-OPENSSL_rdtsc
crypto/alphacpuid.pl:.globl	OPENSSL_rdtsc
crypto/alphacpuid.pl:.ent	OPENSSL_rdtsc
crypto/alphacpuid.pl:OPENSSL_rdtsc:
crypto/alphacpuid.pl:.end	OPENSSL_rdtsc
crypto/armcap.c:unsigned long OPENSSL_rdtsc(void)
crypto/armcap.c:unsigned long OPENSSL_rdtsc(void)
crypto/x86cpuid.pl:&function_begin_B("OPENSSL_rdtsc","EXTRN\t_OPENSSL_ia32cap_P:DWORD");
crypto/x86cpuid.pl:&function_end_B("OPENSSL_rdtsc");
crypto/ia64cpuid.S:.global	OPENSSL_rdtsc#
crypto/ia64cpuid.S:.proc	OPENSSL_rdtsc#
crypto/ia64cpuid.S:OPENSSL_rdtsc:
crypto/ia64cpuid.S:.endp   OPENSSL_rdtsc#
crypto/ppccpuid.pl:.globl	.OPENSSL_rdtsc
crypto/ppccpuid.pl:.OPENSSL_rdtsc:
crypto/ppccpuid.pl:.size	.OPENSSL_rdtsc,.-.OPENSSL_rdtsc
crypto/sparcv9cap.c:unsigned long OPENSSL_rdtsc(void)
crypto/s390xcpuid.S:.globl	OPENSSL_rdtsc
crypto/s390xcpuid.S:.type	OPENSSL_rdtsc,@function
crypto/s390xcpuid.S:OPENSSL_rdtsc:
crypto/s390xcpuid.S:.size	OPENSSL_rdtsc,.-OPENSSL_rdtsc
```
so except the declaraion in crypto/modes/gcm128.c which is plain wrong
and only used in a false-conditional selftest code,  no other
uses from C seem to exist.  The issue is just that missing declaration
where the OPENSSL_rdtsc is defined in crypto/armcap.c

So the prototype with "unsigned long" matches the implementation.

Reply:
Okay, I picked the wrong version.